beacon_node/gossipsub/src/behaviour.rs

```diff
@@ -74,6 +78,9 @@ use std::{cmp::Ordering::Equal, fmt::Debug};
 #[cfg(test)]
 mod tests;
 
+/// IDONTWANT Cache capacity.
+const IDONTWANT_CAP: usize = 100;
```

Comment:
As I write this, I'm potentially regretting it, but maybe we want it configurable. 

Just thinking out loud here:
With an LRUCache, I imagine the cache is always going to be full, its probably not worth an LRUTimeCache because the size is quite small. 

The IDONTWANT's are supposed to be to prevent message duplication and on our networks we can see up to like 2s lags on some messages. 

In terms of topics we use, here are theoretical messages per slot (12s):
- aggregates: 800
- blocks: 1
- each subnet: 488

A lot of these come in bursts. I'd imagine each peer would send IDONTWANT's for each message, which would saturate this cache pretty quickly making it ineffective. 

I think we need it to be the order of the messages we would see in total for maybe 1 slot (12 seconds). This number for us is dynamic too, which making it configurable might make sense, or we could make it time-based. 

Maybe the best way would be to make it an LruTimeCache which has a space and time limit. i.e max 10,000 entries so we can't get dos'd. If message id is 256 bytes. The worst memory usage would be like 2mb per peer. And if we time-bound them to like 1 or 3 seconds it would help handle the bursts and we'd prune the cache back down regularly. 

We could use this data structure, which is pretty efficient if I recall: https://github.com/sigp/discv5/blob/master/src/lru_time_cache.rs